plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py

```diff
@@ -133,11 +133,42 @@ def takedown_request(self, alert_id: str, target: str) -> dict:
         )
 
     def get_alerts(self, alert_params: AlertParams) -> list:
-        return self.make_request("GET", "public/v1/data/alerts/alerts-list", params=alert_params.to_dict()).json()
+        response = self.make_request("GET", "public/v1/data/alerts/alerts-list", params=alert_params.to_dict())
+        try:
+            if response.text:
+                return response.json()
+        except json.decoder.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=e)
+
+        return []
 
     def add_manual_alert(self, manual_alert_params: ManualAlertParams) -> str:
         return self.make_request("PUT", "public/v1/data/alerts/add-alert", json_data=manual_alert_params.to_dict()).text
 
+    def get_cve(self, cve_ids: [str]) -> list:
```

Comment:
maybe the code here could be commented? was personally a bit hard to follow, but if no one else agrees i understand

Reply:
We tried to create this code as reusable, but if it is not clear enough we can simplify it a little bit. Is it ok for you?